### PR TITLE
Adding additional datetime operations, and improving test coverage.

### PR DESCRIPTION
**Pull request by JovanVeljanoski**
Adding additional `datetime` operations. Each operation is covered by a unit-test.

### CODE REVIEW

File: packages/vaex-core/vaex/functions.py [hunk 1]
```diff
@@ -158,12 +158,42 @@ def dt_dayofyear(x):
     # x = x.astype("<M8[ns]")
     return pd.Series(x).dt.dayofyear.values
 
+@register_function(scope='dt', as_property=True)
+def dt_is_leap_year(x):
+    import pandas as pd
+    # x = x.astype("<M8[ns]")
```
Comment (maartenbreddels) [inline packages/vaex-core/vaex/functions.py]: copy paste artifacts.

File: tests/datetime_test.py [hunk 2]
```diff
@@ -1,10 +1,27 @@
...
+
+    assert df.date.dt.hour.values.tolist() == pandas_df.date.dt.hour.values.tolist()
+    assert df.date.dt.minute.values.tolist() == pandas_df.date.dt.minute.values.tolist()
+    assert df.date.dt.second.values.tolist() == pandas_df.date.dt.second.values.tolist()
+    assert df.date.dt.day.values.tolist() == pandas_df.date.dt.day.values.tolist()
+    assert df.date.dt.day_name.values.tolist() == pandas_df.date.dt.day_name().values.tolist()
+    assert df.date.dt.month.values.tolist() == pandas_df.date.dt.month.values.tolist()
+    assert df.date.dt.month_name.values.tolist() == pandas_df.date.dt.month_name().values.tolist()
+    assert df.date.dt.year.values.tolist() == pandas_df.date.dt.year.values.tolist()
+    assert df.date.dt.is_leap_year.values.tolist() == pandas_df.date.dt.is_leap_year.values.tolist()
```
Comment (maartenbreddels) [inline tests/datetime_test.py]: does one of them contain a leapyear?
you could do:
assert any(df.date.dt.is_leap_year.values.tolist())

File: packages/vaex-core/vaex/functions.py (hunk 1 above)
Comment (JovanVeljanoski) [reply packages/vaex-core/vaex/functions.py]: Ok, I've cleaned this up now.

File: tests/datetime_test.py (hunk 2 above)
Comment (JovanVeljanoski) [reply tests/datetime_test.py]: Good point. I made one of them to contain a leap year. I still find it better to just do a direct comparison across all elements.

Comment (maartenbreddels) [reply tests/datetime_test.py]: I mean both. 
```python
assert df.date.dt.is_leap_year.values.tolist() == pandas_df.date.dt.is_leap_year.values.tolist()
assert any(df.date.dt.is_leap_year.values.tolist())
```

Comment (JovanVeljanoski) [reply tests/datetime_test.py]: Ok, added both.

**Comment by maartenbreddels** [comment]
Actually, you can remove all these!
```
    @property
    def day_name(self):
        return self.expression.ds.func.dt_day_name(self.expression)
```

`@register_function(scope='dt', as_property=True)` will add these automatically (look at the stringoperations class!)

I almost forgot.

**Comment by JovanVeljanoski** [comment]
I removed the redundant code. Tests pass.

**Pull request merged**